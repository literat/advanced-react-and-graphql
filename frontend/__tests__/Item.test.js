import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: 'ABC123',
  title: 'A Cool Item',
  price: 5000,
  description: 'This item is really cool!',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg',
};

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  /* it('renders the image properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const img = wrapper.find('img');

    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it('renders the price tag and title properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = wrapper.find('PriceTag');

    expect(PriceTag.children().text()).toBe('$50');
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  });

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonsList = wrapper.find('.buttonList');
    expect(buttonsList.children()).toHaveLength(3);
    expect(buttonsList.find('Link')).toHaveLength(1);
    expect(buttonsList.find('AddToCart').exists()).toBe(true);
    expect(buttonsList.find('DeleteItem').exists()).toBe(true);
  }); */
});
