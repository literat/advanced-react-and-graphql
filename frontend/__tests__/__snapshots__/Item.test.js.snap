// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<div>
  <div
    class="ItemStyles__Item-sc-16pk14u-0 EJjXi"
  >
    <img
      alt="dogs are best"
      src="dog-small.jpg"
    />
    <h3
      class="Title-sc-16nq74k-0 iuvmfe"
    >
      <a
        href="/item?id=abc123"
      >
        dogs are best
      </a>
    </h3>
    <span
      class="PriceTag-nwbk6t-0 gwFXfj"
    >
      $50
    </span>
    <p>
      dogs
    </p>
    <div
      class="buttonList"
    >
      <a
        href="/update?id=abc123"
      >
        Edit ✏️
      </a>
      <button
        type="button"
      >
        Add
         To Cart 🛒
      </button>
      <button
        type="button"
      >
        Delete This Item
      </button>
    </div>
  </div>
</div>
`;
